src/physics/em/detail/SBEnergyDistribution.hh

```diff
@@ -0,0 +1,141 @@
+ * the scaled cross section \f$ \chi \f$.
+ * The analytic sample over the domain is from \f[
+   p_1(\kappa) \frac{1}{\ln 1/\kappa_c} \frac{1}{\kappa}
+ * \f]
+ * by sampling
+ * \f[
+   \kappa = \exp( \xi \ln \kappa_c ) \,,
+   \f]
+ * and the rejection sample is the ratio of the scaled differential cross
+ * section at the exiting energy to the maximum across all exiting energies.
+ * \f[
+   p_2(\kappa) = \frac{\chi_Z(E, \kappa)}{\max_\kappa \chi_Z(E, \kappa)}
+   \f]
+ * Since the tabulated data is bilinearly interpolated in incident and exiting
+ * energy, we can calculate a bounding maximum by precalculating (at
+ * setup time) the index of the maximum value of \f$ \chi \f$ and
+ * linearly interpolating those maximum values based on the incident energy.
+ *
+ * The above algorithm is idealized; in practice, the minimum and maximum
+ * values are adjusted for a constant factor \f$d_\rho\f$, which depends on the
+ * incident particle mass + energy and the material's electron density:
+ * \f[
+    \kappa_\mathrm{min} = \ln (E_c^2 + d_\rho E^2)
+ * \f]
+ * and
+ * \f[
+    \kappa_\mathrm{max} = \ln (E^2 + d_\rho E^2) \, .
+ * \f]
+ * With this correction, the sample of the exiting gamma energy
+ * \f$ k = \kappa / E \f$ is transformed from the simple exponential above to:
+ * \f[
+ *  k = \sqrt{ \exp(\kappa_\mathrm{min} + \xi [\kappa_\mathrm{max} -
+ \kappa_\mathrm{min}]) - d_\rho E^2}
+ * \f]
+ *
+ * \todo Add a template parameter for a cross section scaling, which will be an
+ * identity function (always returning 1) for electrons but is
+ * SBPositronXsScaling for positrons. Use it to adjust max_xs on construction
+ * and modify the sampled xs during iteration.
+ */
+class SBEnergyDistribution
+{
+  public:
+    //!@{
+    //! Type aliases
+    using SBData
+        = SeltzerBergerData<Ownership::const_reference, MemSpace::native>;
+    using Energy   = units::MevEnergy;
+    using EnergySq = Quantity<UnitProduct<units::Mev, units::Mev>>;
+    //!@}
+
+  public:
+    // Construct from data
+    inline CELER_FUNCTION SBEnergyDistribution(const SBData& data,
+                                               Energy        inc_energy,
+                                               ElementId     element,
+                                               EnergySq density_correction,
+                                               Energy   min_gamma_energy);
+
+    template<class Engine>
+    inline CELER_FUNCTION Energy operator()(Engine& rng);
+
+    //// DEBUG FUNCTIONALITY ////
+
+    //! Maximum cross section calculated for rejection
+    real_type max_xs() const { return 1 / inv_max_xs_; }
+
+  private:
+    //// IMPLEMENTATION DATA ////
+
+    using SBTables
+        = SeltzerBergerTableData<Ownership::const_reference, MemSpace::native>;
+    using UniformSampler = UniformRealDistribution<>;
+
+    const real_type             inc_energy_;
+    const TwodSubgridCalculator calc_xs_;
```

Comment:
I am a bit confused by **const** here.  Is the life time of this class for each track (i.e. instantiated for each track)?

Reply:
The lifetime of the class is going to be inside the `SBInteractor::operator()`. It's constructed after the incident particle energy and material are known, and (with the exception of unit tests) used only once and discarded. The `const` here is just to indicate that these properties are fixed during construction and not set later (or reused during sampling).